repos/system_upgrade/el8toel9/actors/checksaphana/libraries/checksaphana.py

```diff
@@ -0,0 +1,218 @@
+    found = {}
+    for instance in info.instances:
+        if _manifest_get(instance.manifest, 'release') == '1.00':
+            _add_hana_details(found, instance)
+
+    if found:
+        detected = _create_detected_instances_list(found)
+        reporting.create_report([
+            reporting.Title('Found SAP HANA 1 which is not supported with the target version of RHEL'),
+            reporting.Summary(
+                ('SAP HANA 1.00 is not supported with the version of RHEL you are upgrading to.\n\n'
+                 'The following instances have been detected to be version 1.00:\n'
+                 '{}'.format(detected))
+            ),
+            reporting.Severity(reporting.Severity.HIGH),
+            reporting.RemediationHint((
+                'In order to upgrade RHEL, you will have to upgrade your SAP HANA 1.0 software to '
+                '{supported}.'.format(supported=SAP_HANA_MINIMAL_VERSION_STRING))),
+            reporting.ExternalLink(url='https://launchpad.support.sap.com/#/notes/2235581',
+                                   title='SAP HANA: Supported Operating Systems'),
+            reporting.Groups([reporting.Groups.SANITY]),
+            reporting.Groups([reporting.Groups.INHIBITOR]),
+            reporting.Audience('sysadmin')
+        ])
+
+
+def _major_version_check(instance):
+    """ Performs the check for the major version of SAP HANA """
+    release = _manifest_get(instance.manifest, 'release', '0.00')
+    parts = release.split('.')
+
+    try:
+        if int(parts[0]) != SAP_HANA_MINIMAL_MAJOR_VERSION:
+            api.current_logger().info('Unsupported major version {} for instance {}'.format(release, instance.name))
+            return False
+        return True
+    except (ValueError, IndexError):
+        api.current_logger().warn(
+            'Failed to parse manifest release field for instance {}'.format(instance.name), exc_info=True)
+        return False
+
+
+def _sp_rev_patchlevel_check(instance):
+    """ Checks whether this SP, REV & PatchLevel are eligible """
+    number = _manifest_get(instance.manifest, 'rev-number', '000')
+    if len(number) > 2 and number.isdigit():
+        required_sp_levels = [r[0] for r in SAP_HANA_RHEL8_REQUIRED_PATCH_LEVELS1]
+        lowest_sp = min(required_sp_levels)
+        sp = int(number[0:2].lstrip('0') or '0')
+        if sp < lowest_sp:
+            # Less than minimal required SP
+            return False
+        for requirements in SAP_HANA_RHEL8_REQUIRED_PATCH_LEVELS1:
+            req_sp, req_rev, req_pl = requirements
+            if sp == req_sp:
+                rev = int(number.lstrip('0') or '0')
+                if rev < req_rev:
+                    continue
+                if rev == req_rev:
+                    patch_level = int(_manifest_get(instance.manifest, 'rev-patchlevel', '00').lstrip('0') or '0')
+                    if patch_level < req_pl:
+                        continue
+                return True
```

Comment:
Please make this a separate function and not duplicate it

Reply:
@vinzenz I know it's terrible, but we tested it already and the deadline for the whole thing to make it in time for a release is today. Ideally, I'd like to move running check to common actors, and keep versions check in corresponding repos paths. May we keep it as it is for now, and I will change it a bit later? 
I can't figure out how to make it with unit tests to pass